HowToBuild.md

```diff
@@ -0,0 +1,12 @@
+# How to build this package
+
+## Build and Test
+
+```sh
+R CMD build .                           # creates e.g. tiledbcloud_0.0.4.tar.gz
+R CMD check tiledbcloud_0.0.4.tar.gz    # checks the package
+```
+
+## Demo
+
+A few first commands: see [local/](local/).
```

Comment:
These will be expanded in an upcoming PR. As is, they don't yet reflect work done between 0.0.3 and 0.0.4.